build/projects/Add-Ons (W1)/.AL-Go/PreCompileApp.ps1

```diff
@@ -0,0 +1,20 @@
+Param(
+    [string] $appType,
+    [ref] $compilationParams
+)
+
+if ($ENV:BuildMode -eq 'Clean') {
+
+    # If useProjectDependencies is set to false, we need to remove the app symbols folder so we recompile everything.
+    $settings = Get-Content (Join-Path $PSScriptRoot "settings.json" -Resolve) | ConvertFrom-Json
+    if ($settings.useProjectDependencies -eq $false) {
+        $symbolsFolder = $compilationParams.Value["appSymbolsFolder"]
+        if (Test-Path $symbolsFolder) {
+            Remove-Item -Path $symbolsFolder\* -Recurse -Force -Verbose
```

Comment:
Wouldn't that mean that the external dependencies will be recompiled for every add-on app? There should be a smarter way to recompile them once.

Reply:
True 😄 That's not ideal. I'll see if I can address that